packages/arb-validator-core/gobridge/globalInbox.go

```diff
@@ -49,33 +51,35 @@ func (ib *inbox) addMessageToInbox(msg common.Hash) {
 }
 
 func deployGlobalInbox(m *goEthdata) {
-	m.globalInboxContract = &globalInbox{
+	m.globalInbox = &globalInbox{
+		ethData:         m,
 		inbox:           make(map[common.Address]*inbox),
 		contractAddress: m.getNextAddress(),
 	}
 }
 
 func newGlobalInbox(address common.Address, client *GoArbAuthClient) (*globalInbox, error) {
-	client.globalInboxContract.inbox[address] = &inbox{
-		value:  value.NewEmptyTuple().Hash(),
-		count:  big.NewInt(0),
-		client: client,
+	client.globalInbox.inbox[address] = &inbox{
+		value: value.NewEmptyTuple().Hash(),
+		count: big.NewInt(0),
 	}
-	return client.globalInboxContract, nil
+	return client.globalInbox, nil
 }
```

Comment:
The address passed here is the address of the global inbox. It doesn't make sense to initialize an inbox with that same address.

Reply:
fixed